### PR TITLE
feat: Allow `Enum` for `SelectBone`-values and implement `User`s status as `Enum` 

### PR DESCRIPTION
**Pull request by sveneberth**
Often I've something like:
```py
class MySkel(Skeleton):
    status = SelectBone(
        descr="Status",
        values={
            "active": "Active",
            "disabled": "Disabled",
            "pending": "Pending",
        },
    )
```
Than I do something:
```py
skel = MySkel()
if skel["status"] == "pending":
    # do some magic
    ...
    skel["status"] = "active"
```
But this is very vulnerable for typos and bad for refactoring.
Therefore I implemented the support of an `Enum` for `SelectBone` values.

As the first usage I implement the `User` status as `Enum`.

### CODE REVIEW

**Comment by ArneGudermann** [comment]
Hey @sveneberth thank you for this PR. But I think your code is not backwards compatibility when i have old code like this
```python
cuser = current.user.get()
if cuser["status"] == 10: # This will fail
	print("ok")
if cuser["status"] > 10: # This will crash
	print("ok")
```

**Comment by sveneberth** [comment]
> But I think your code is not backwards compatibility when i have old code like this

Hmm. Indeed, this is not 100% downward-compliant. However, the question is how many projects really access the status or if the amount of customization would be justifiable.

Let's best talk about it in the ViUR-meeting.

Otherwise we can reduce this PR for the time being only to the optional `Enum` functionality and replace this in the `UserSkel` at a later release with other breaking changes.

**Comment by phorward** [comment]
> > But I think your code is not backwards compatibility when i have old code like this
> 
> Hmm. Indeed, this is not 100% backwards compatible. However, the question is how many projects really access the status or if the amount of customization would be justifiable.

This problem is solvable!

You can define the `Status` enum this way, to resolve @ArneGudermann's annotated problem:
```python
class Status(enum.Enum):
    WAITING_FOR_EMAIL_VERIFICATION = 1  # Waiting for email verification
    WAITING_FOR_ADMIN_VERIFICATION = 2  # Waiting for verification through admin
    DISABLED = 5  # Account disabled
    ACTIVE = 10  # Active

    def __eq__(self, other):
        return self.value == other

    def __ne__(self, other):
        return self.value != other

    def __lt__(self, other):
        return self.value < other

    def __gt__(self, other):
        return self.value > other

    def __le__(self, other):
        return self.value <= other

    def __ge__(self, other):
        return self.value >= other
```

The much smarter solution would be to provide a `CompareableEnum`-class in the `SelectBone` module, implementing this functionality, and let `Status` inherit from this one. This allows to use the feature if wanted, and generalizes it as far as possible.

**Comment by sveneberth** [comment]
> This problem is solvable!

Yeah, of course it is :smiley: 

Sooner or later this should be adjusted anyway. Just like assignments like `user_skel["status"] = 10`.
I don't really care. But I don't think that every change to components should be backwards compatible if it doesn't belong to the standard API area. It all just bloats the code unnecessarily.

**Comment by phorward** [comment]
>

**Comment by phorward** [comment]
> Sooner or later this should be adjusted anyway. Just like assignments like `user_skel["status"] = 10`. I don't really care. But I don't think that every change to components should be backwards compatible if it doesn't belong to the standard API area. It all just bloats the code unnecessarily.

Hello rage-mode-sven!

Why not simply apply my proposal? I don't think that 19 lines of code (including empty lines) bloat something when it is fully backward-compatible. In viur-core 3.x we should be backward-compatible as much as possible. This is important for projects acceptance and fun, also in very, very tiny communities. Any maybe, the possibility to compare for lower might be useful, so why not offer it? 

Please just add my proposal from above to your PR and it can be merged, as it is a nice extension. Thanx.

**Comment by sveneberth** [comment]
> Hello rage-mode-sven!
> 
> Why not simply apply my proposal? I don't think that 19 lines of code (including empty lines) bloat something when it is fully backward-compatible. In viur-core 3.x we should be backward-compatible as much as possible. This is important for projects acceptance and fun, also in very, very tiny communities. Even, maybe, the possibility to compare for lower status values might be useful, so why not offer it using the proposed `CompareableEnum`-class to inherit from?
> 
> Please just add my proposal from above to your PR and it can be merged, as it is a nice extension. Thanx.

All right sir maintainer @phorward. As commanded I have made the changes.

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @sveneberth, this is better, can you please take a look on my comments?

File: core/modules/user.py [hunk 1]
```diff
@@ -19,6 +21,30 @@
...
+    Has backwards compatibility to be comparable with non-enum values.
+    Will be removed with viur-core 4.0.0
+    """
+
+    WAITING_FOR_EMAIL_VERIFICATION = 1  # Waiting for email verification
+    WAITING_FOR_ADMIN_VERIFICATION = 2  # Waiting for verification through admin
+    DISABLED = 5  # Account disabled
+    ACTIVE = 10  # Active
+
+    def __eq__(self, other):
```
[the remark this answers is not in the document]
Comment (sveneberth) [reply core/modules/user.py:37]: They are provided by [`@functools.total_ordering`](https://docs.python.org/3/library/functools.html#functools.total_ordering)

[the remark this answers is not in the document]
Comment (phorward) [reply core/modules/user.py:37]: Ahh, thanks for clarification!

**Review by phorward — APPROVED** [review]

**Pull request merged**